### PR TITLE
Budget Main Screen

### PR DESCRIPTION
**Pull request by DanWilkins2107**

### CODE REVIEW

File: components/Budget/ImageViewer.jsx
Comment (dwulfse) [inline components/Budget/ImageViewer.jsx:1]: help me understand what RecieptViewer and ImageViewer are

File: components/Budget/ExpenseSummary.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                            textSize="2xl"
+                            loadingSize="small"
+                        />
+                        <View className="ml-2">
+                            <Subtitle extraClassName={expense.placeholder && "text-gray-600"}>
+                                {expense.reference}
+                            </Subtitle>
+                        </View>
+                    </View>
+                    <View className="flex-row items-center">
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx]: check overflow

File: components/Budget/ExpenseSummary.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                        {isUser && <TouchableOpacity
+                            className={`w-10 h-10 rounded-md border-2 mr-4 justify-center items-center ${
+                                expense.placeholder && "border-gray-600"
+                            }`}
+                            onPress={() => {
+                                alert("Implement editing expense");
+                            }}
+                        >
+                            <Icon
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx]: try line up the edit icons

File: screens/ProductionScreens/BudgetMain.jsx [hunk 3]
```diff
@@ -23,39 +37,138 @@ export default function BudgetMainScreen({ route }) {
...
                     })
                 );
                 setExpenses(expenseData);
+                setLoading(false);
             });
         };
         fetchData();
     }, []);
 
+    const filterExpenses = (expenses) => {
```
Comment (dwulfse) [inline screens/ProductionScreens/BudgetMain.jsx:56]: check time ordering

File: screens/ProductionScreens/BudgetMain.jsx [hunk 4]
```diff
@@ -23,39 +37,138 @@ export default function BudgetMainScreen({ route }) {
...
+                                        />
+                                    );
+                                })}
                             </>
                         )}
-                    </View>
-                );
-            })}
-        </ScrollView>
+                    </ScrollView>
```
Comment (dwulfse) [inline screens/ProductionScreens/BudgetMain.jsx:208]: lil spacer at the bottom

File: screens/ProductionScreens/BudgetMain.jsx [hunk 5]
```diff
@@ -23,39 +37,138 @@ export default function BudgetMainScreen({ route }) {
...
+                            <Checkbox
+                                checked={isPlaceholder}
+                                setChecked={() => setIsPlaceholder(!isPlaceholder)}
+                                size={20}
+                            />
+                            <Subtitle extraClassName="ml-2">Include Placeholders?</Subtitle>
+                        </View>
+                    </View>
+                    <View className="border-b-2 w-[90%]">
+                        <Title extraClassName="mb-4 text-center">Expenses</Title>
```
Comment (dwulfse) [inline screens/ProductionScreens/BudgetMain.jsx:129]: if you can find some more space at the top that would be beautiful

**Review by dwulfse — CHANGES_REQUESTED** [review]

File: components/Budget/ImageViewer.jsx
Comment (DanWilkins2107) [reply components/Budget/ImageViewer.jsx:1]: RecieptViewer is the component for adding a receipt to an expense. RecieptViewerModal is the modal that pops up whenever we click on an image. ImageViewer is the picture in the dropdown when viewing an expense.

File: screens/ProductionScreens/BudgetMain.jsx (hunk 5 above)
Comment (DanWilkins2107) [reply screens/ProductionScreens/BudgetMain.jsx:129]: Honestly I don't have a clue what I could remove lol
Feel like if I get rid of any margin/padding it'll just look too cramped

File: screens/ProductionScreens/BudgetMain.jsx [hunk 6]
```diff
@@ -23,39 +38,166 @@ export default function BudgetMainScreen({ route }) {
...
+                            {filtersOpen && (
+                                <View className="flex-col">
+                                    <View className="flex-row justify-start items-center px-2 mt-2">
+                                        <Checkbox
+                                            checked={hidePlaceholders}
+                                            setChecked={() =>
+                                                setHidePlaceholders(!hidePlaceholders)
+                                            }
+                                        />
+                                        <Subtitle extraClassName="ml-4">
```
Comment (dwulfse) [inline screens/ProductionScreens/BudgetMain.jsx:162]: check box on text press

File: components/Budget/ExpenseSummary.jsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                                <Text className="font-semibold text-base mt-2">Description:</Text>
+                                <Text>{expense.description}</Text>
+                                <Text className="font-semibold text-base mt-2">Submitted:</Text>
+                                <Text>{timeString}</Text>
+                            </View>
+                            {expense.receipt && (
+                                <View className="flex-1 justify-center items-center">
+                                    <Text className="font-semibold text-base text-center mb-2">
+                                        Receipt:
+                                    </Text>
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx]: remove colon

File: components/Budget/ExpenseSummary.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        <View className="py-1">
+            <TouchableOpacity
+                className={`justify-between w-full ${
+                    expense.placeholder ? "bg-slate-50" : "bg-slate-200"
+                } rounded-lg p-2 border-2`}
+                onPress={() => setIsDropdownOpen(!isDropdownOpen)}
+            >
+                <View className="flex-row justify-between">
+                    <View className="flex-row items-center flex-1">
+                        <ProfilePicture
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx:22]: can this get a grey outline for placeholder

Comment (dwulfse) [reply components/Budget/ExpenseSummary.jsx:22]: think being able to change outline colour is a good shout generally, could do some things with diff outlines for admins n that

File: components/Budget/ExpenseSummary.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+                                <Text className="font-semibold text-base mt-2">Submitted:</Text>
+                                <Text>{timeString}</Text>
+                            </View>
+                            {expense.receipt && (
+                                <View className="flex-1 justify-center items-center">
+                                    <Text className="font-semibold text-base text-center mb-2">
+                                        Receipt:
+                                    </Text>
+                                    <View className="w-40 h-40">
+                                        <ImageViewer URL={expense.receipt} centered />
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx:75]: think the spacing around the reciept looks slightly odd- too much at the side/ not enough at the bottom

File: components/Budget/BudgetLineGraph.jsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    const placeholderPercentage = (valueToUse / budget) * 100;
+    const isOverBudget = placeholderPercentage > 100;
+    const budgetPercentage = isOverBudget ? 100 : placeholderPercentage;
+
+    const animatedValue = useSharedValue(0);
+
+    const animatedStyle = useAnimatedStyle(() => {
+        const backgroundColor = interpolateColor(
+            animatedValue.value,
+            [0, 33, 75, 100],
```
Comment (dwulfse) [inline components/Budget/BudgetLineGraph.jsx]: play with where the green ends

File: components/Budget/ExpenseSummary.jsx [hunk 11]
```diff
@@ -72,9 +72,9 @@ export default function ExpenseSummary({ expense, isUser }) {
                             {expense.receipt && (
                                 <View className="flex-1 justify-center items-center">
                                     <Text className="font-semibold text-base text-center mb-2">
-                                        Receipt:
+                                        Receipt
```
Comment (dwulfse) [inline components/Budget/ExpenseSummary.jsx]: remove "receipt" completely

**Review by dwulfse — APPROVED** [review]

**Pull request merged**